modules/core/src/main/java/org/apache/ignite/internal/processors/platform/PlatformContextImpl.java

```diff
@@ -80,10 +83,13 @@
  * Implementation of platform context.
  */
 @SuppressWarnings("TypeMayBeWeakened")
-public class PlatformContextImpl implements PlatformContext {
+public class PlatformContextImpl implements PlatformContext, PartitionsExchangeAware {
     /** Supported event types. */
     private static final Set<Integer> evtTyps;
 
+    /** Whether to use thread-local data to update platform near cache. */
+    private static final ThreadLocal<Boolean> nearUpdateUseThreadLocal = new ThreadLocal<>();
```

Comment:
Do we use TLS here to avoid redundant copies on put from platform side? Looks a bit complicated.

Reply:
Yes, when .NET performs `Put`, we already have a value on .NET side, no need to send it back through the callback. 

Before this TLS optimization we had a **50%** perf drop on `Put` when Platform Near is enabled. After TLS optimization the drop is about **7%**. I think this added complexity is worth it.

Reply:
Cool